frontend-web/webclient/app/Project/Members.tsx

```diff
@@ -551,494 +486,162 @@ export const ProjectMembers2: React.FunctionComponent = () => {
-                            <Flex width={"100%"}>
-                                <Truncate fontSize="25px" width={1}>{inspectingGroup.specification.title}</Truncate>
-                            </Flex>
-                        </Flex>
-
-                        {inspectingGroup.status.members!.length !== 0 ? null : <>
-                            <Text mt={40} textAlign="center">
-                                <Heading.h4>No members in group</Heading.h4>
-                                You can add members by clicking on the green arrow in the
-                                &apos;Members&apos; panel.
-                            </Text>
-                        </>}
-
-                        <List>
-                            {inspectingGroup.status.members!.map((member, idx) => (
-                                <ItemRow
-                                    item={member}
-                                    key={idx}
-                                    browseType={BrowseType.Embedded}
-                                    renderer={GroupMemberRenderer}
-                                    toggleSet={groupMemberToggleSet}
-                                    operations={groupMemberOperations}
-                                    callbacks={callbacks}
-                                    itemTitle={"Member"}
-                                />
-                            ))}
-                        </List>
-                    </>}
-                </div>
-            </Box>
-        }
-    />;
-};
-
-// Secondary interface (e.g. member rows and group rows)
-// ================================================================================
-const MemberRenderer: ItemRenderer<ProjectMember, Callbacks> = {
-    Icon: ({resource}) => {
-        const avatars = useAvatars();
-        if (!resource) return null;
-
-        return <UserAvatar avatar={avatars.avatar(resource.username)} height="48px" width="48px" />;
-    },
-
-    MainTitle: ({resource}) => {
-        if (!resource) return null;
-        return <Text fontSize={"16px"}>{resource.username}</Text>;
-    },
-
-    ImportantStats: ({resource, callbacks}) => {
-        const {inspectingGroup, isAdmin, project} = callbacks;
-        if (!resource || !project) return null;
+    const [inviteLinks, setInviteLinks] = useState<ProjectInviteLink[]>([]);
 
-        const memberOfThisGroup = !inspectingGroup ? false : inspectingGroup.status.members!.some(m => m === resource.username);
-
-        let options: RoleOption[];
-        if (!isAdmin || resource.username === Client.username || !!inspectingGroup) {
-            options = [roleOptionForRole(resource.role)];
-        } else {
-            options = [roleOptionForRole(OldProjectRole.USER), roleOptionForRole(OldProjectRole.ADMIN)];
-
-            if (project.status.myRole! === OldProjectRole.PI) {
-                options.push(roleOptionForRole(OldProjectRole.PI));
-            }
+    useEffect(() => {
+        let didCancel = false;
```

Comment:
I think this may have the issue of being captured as a value in the function definition.

`const cancellation = {didCancel: false};` and then later `if (!cancellation.didCancel)` should fix it.